src/protocol/payload/reject.rs

```diff
@@ -0,0 +1,93 @@
+use crate::protocol::payload::{read_n_bytes, VarStr};
+
+use std::io::{self, Cursor, Read, Write};
+
+#[derive(Debug)]
+pub struct Reject {
+    message: VarStr,
+    ccode: CCode,
+    reason: VarStr,
+    data: Vec<u8>,
+}
+
+impl Reject {
+    pub fn encode(&self, buffer: &mut Vec<u8>) -> io::Result<()> {
+        self.message.encode(buffer)?;
+        self.ccode.encode(buffer)?;
+        self.reason.encode(buffer)?;
+        buffer.write_all(&self.data)
+    }
+
+    pub fn decode(bytes: &mut Cursor<&[u8]>) -> io::Result<Self> {
+        let message = VarStr::decode(bytes)?;
+        let ccode = CCode::decode(bytes)?;
+        let reason = VarStr::decode(bytes)?;
+
+        let mut data = Vec::new();
+        bytes.read_to_end(&mut data)?;
+
+        Ok(Self {
+            message,
+            ccode,
+            reason,
+            data,
+        })
+    }
+}
+
+const MALFORMED_CODE: u8 = 0x01;
+const INVALID_CODE: u8 = 0x10;
+const OBSELETE_CODE: u8 = 0x11;
+const DUPLICATE_CODE: u8 = 0x12;
+const NON_STANDARD_CODE: u8 = 0x40;
+const DUST_CODE: u8 = 0x41;
+const INSUFFICIENT_FEE_CODE: u8 = 0x42;
+const CHECKPOINT_CODE: u8 = 0x43;
```

Comment:
Zebra seems to have [an extra code](https://github.com/ZcashFoundation/zebra/blob/24bf952e982bde28eb384b211659159d46150f63/zebra-network/src/protocol/external/codec.rs#L504)? Not sure if this is included in zcashd nodes.

Reply:
Its not mentioned in the bitcoin protocol, so might be a zcash addition?